src/App.js

```diff
@@ -14,11 +14,74 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    selectedGoods: [],
+  }
+
+  selectBtnClick = (arr, good) => {
+    arr.push(good);
+    this.setState({ selectedGoods: arr });
+  }
+
+  removeBtnClick = (arr, good) => {
+    arr.splice(arr.indexOf(good), 1);
+    this.setState({ selectedGoods: arr });
+  }
+
+  render() {
+    return (
+      <div className="App">
+        <h1>
+          Selected good: -
+          {this.state.selectedGoods.length}
+        </h1>
+        <h2>
+          {this.state.selectedGoods.length !== 0
+            ? (
+              `${this.state.selectedGoods.join(', ')
+                  + (this.state.selectedGoods.length === 1
+                    ? ' is'
+                    : ' are')} selected`
+            )
+            : 'No goods selected'
+          }
+        </h2>
+        <ul>
+          {goodsFromServer.map(good => (
+            <li key={good}>
+              <span>{good}</span>
+              {
+                  this.state.selectedGoods.indexOf(good) === -1
+                    ? (
+                      <button
+                        type="button"
+                        className="select"
+                        onClick={() => {
+                          this.selectBtnClick(this.state.selectedGoods, good);
```

Comment:
```suggestion
                          this.selectBtnClick(good);
```
Не потрібно передавати масив зі стейту. Ти можеш прямо у функції змінювати цей масив, використовуючи setState